### PR TITLE
[XNIO-422] Add information about remote connection address to the Exc…

### PR DESCRIPTION
**Pull request by spyrkob**
…eption when connection fails

JIRA: https://issues.redhat.com/browse/XNIO-422

### CODE REVIEW

File: nio-impl/src/main/java/org/xnio/nio/WorkerThread.java [hunk 1]
```diff
@@ -335,7 +335,9 @@ void handleReady(final int ops) {
                 }
             } catch (IOException e) {
                 selectorLog.tracef("ConnectHandle.handleReady Exception, %s", e);
-                futureResult.setException(e);
+                futureResult.setException(new IOException(String.format("%s - failed to connect to %s",
```
Comment (fl4via) [inline nio-impl/src/main/java/org/xnio/nio/WorkerThread.java]: @spyrkob please, add the exception message to org.xnio.nioLog

**Review by fl4via — CHANGES_REQUESTED** [review]

File: nio-impl/src/main/java/org/xnio/nio/WorkerThread.java (hunk 1 above)
Comment (spyrkob) [reply nio-impl/src/main/java/org/xnio/nio/WorkerThread.java]: sorry, I missed that. updated now

**Comment by spyrkob** [comment]
Hi @fl4via, I'm struggling to get a clean run of testsuite on my local env - do you know if those failures are expected?

**Comment by dmlloyd** [comment]
See my concerns on #306 - we should not be logging IP addresses in the normal log.

**Comment by spyrkob** [comment]
Thanks @dmlloyd. Is there any alternative way to add this info to make debugging easier?

**Comment by dmlloyd** [comment]
One thing we talked about was "log redaction" where you would set up a `@LogMessage` and pass in the "sensitive" thing as a parameter (password, IP address, that sort of thing) and then we'd "redact" certain things in the log formatter config, by type, so it would replace the string with something like `<redacted>` or `▒▒▒▒▒▒▒` or something. But IIRC that work was never completed.

In the near term what you could do is add a `TRACE` log message which includes the exception with the IP address.

Either way though, putting the IP address into the exception string is not advised.

**Comment by spyrkob** [comment]
thanks @dmlloyd I moved the information to a trace log now

**Comment by bmaxwell** [comment]
The EAP server.log logs its IP as well as the other servers it connects to and failure to connect is already logged in other flows, this is just one where it is not.  passwords, secret keys and stuff would be things  that we should not log.  Also the server.log would be protected by OS file permissions by admins.

**Review by dmlloyd — APPROVED** [review]

**Review by fl4via — APPROVED** [review]

**Pull request merged**